ethereum_history_api/oracles/src/noir/noir_js/encode.ts

```diff
@@ -1,8 +1,14 @@
-import { isHex } from 'viem';
+import { Hex, isHex } from 'viem';
 import { BYTE_HEX_LENGTH } from '../../util/const.js';
 
 // ENCODERS
-export function encodeHexString(value: string): Uint8Array {
+export function encodeHexString(value: string): Hex[] {
+  return Array.from(encodeHexStringToArray(value)).map(
+    (byte) => `0x${byte.toString(16).padStart(BYTE_HEX_LENGTH, '0')}`
+  ) as Hex[];
```

Comment:
This will probably work. We should try to do type casts as close to value as possible to types in a propagation tree are as correct and strict as we can

```suggestion
  return Array.from(encodeHexStringToArray(value)).map(
    (byte) => `0x${byte.toString(16).padStart(BYTE_HEX_LENGTH, '0')}` as Hex
  );
```

Reply:
I tried the casting you're proposing but it results in linter error: This assertion is unnecessary since it does not change the type of the expression. Not sure why